patches/minecraft/net/minecraft/block/AbstractRailBlock.java.patch

```diff
@@ -1,19 +1,24 @@
 --- a/net/minecraft/block/AbstractRailBlock.java
 +++ b/net/minecraft/block/AbstractRailBlock.java
-@@ -13,7 +_,7 @@
+@@ -13,9 +_,10 @@
  import net.minecraft.world.IWorldReader;
  import net.minecraft.world.World;
  
 -public abstract class AbstractRailBlock extends Block {
 +public abstract class AbstractRailBlock extends Block implements net.minecraftforge.common.extensions.IAbstractRailBlock {
     protected static final VoxelShape field_185590_a = Block.func_208617_a(0.0D, 0.0D, 0.0D, 16.0D, 2.0D, 16.0D);
     protected static final VoxelShape field_190959_b = Block.func_208617_a(0.0D, 0.0D, 0.0D, 16.0D, 8.0D, 16.0D);
++   public static net.minecraft.state.EnumProperty<RailShape> SHAPE = net.minecraft.state.properties.BlockStateProperties.field_208165_R;
```

Comment:
Why?

Reply:
This is the default RailShape Property.

Reply:
Yes, but it's making the patch larger for no functional gain, just directly reference BlockStatePropeeties where you use it, and let the modder add this field if they need it.

Reply:
Placing a vanilla rail adjacent to any AbstractRailBlock will try to setValue the shape property of the AbstractRailBlock, without ever checking for the existence of that property. (This results in a NPE for any AbstractRailBlock that does not have a RailShape Property)
The point of this PR is to patch that.